src/hotspot/share/memory/universe.hpp

```diff
@@ -115,11 +89,11 @@ class Universe: AllStatic {
   // preallocated cause message for delayed StackOverflowError
   static OopHandle    _delayed_stack_overflow_error_message;
 
-  static LatestMethodCache* _finalizer_register_cache; // static method for registering finalizable objects
-  static LatestMethodCache* _loader_addClass_cache;    // method for registering loaded classes in class loader vector
-  static LatestMethodCache* _throw_illegal_access_error_cache; // Unsafe.throwIllegalAccessError() method
-  static LatestMethodCache* _throw_no_such_method_error_cache; // Unsafe.throwNoSuchMethodError() method
-  static LatestMethodCache* _do_stack_walk_cache;      // method for stack walker callback
+  static LatestMethodCache _finalizer_register_cache; // static method for registering finalizable objects
+  static LatestMethodCache _loader_addClass_cache;    // method for registering loaded classes in class loader vector
+  static LatestMethodCache _throw_illegal_access_error_cache; // Unsafe.throwIllegalAccessError() method
+  static LatestMethodCache _throw_no_such_method_error_cache; // Unsafe.throwNoSuchMethodError() method
+  static LatestMethodCache _do_stack_walk_cache;      // method for stack walker callback
```

Comment:
Does this compile with all compilers with the forward declaration if the type isn't a pointer?  You could further reduce this by moving these static declarations into universe.cpp.  I don't see why they have to be exposed here.